python/astro_metadata_translator/translators/suprimecam.py

```diff
@@ -119,6 +119,15 @@ def to_datetime_end(self):
         value = self._from_fits_date_string(self._header["DATE-OBS"],
                                             time_str=self._header["UT-END"], scale="utc")
         self._used_these_cards("DATE-OBS", "UT-END")
+
+        # Sometimes the end time is less than the begin time plus the
+        # exposure time so we have to check for that.
+        exposure_time = self.to_exposure_time()
+        datetime_begin = self.to_datetime_begin()
+        exposure_end = datetime_begin + exposure_time
+        if value < exposure_end:
+            value = exposure_end
```

Comment:
Shouldn't this be an error, or is it actually expected?

Reply:
There seems to be a problem with Suprimecam data where the end time in the header is less than the start time plus the exposure time. I don't have enough suprimecam data to know if that's always true so this test is there to ensure that the the latest end time is used.